contracts/InterestRate/StableRateModel.sol

```diff
@@ -0,0 +1,135 @@
+
+    uint256 private constant BASE = 1e18;
+
+    /// @notice The approximate number of blocks per year that is assumed by the interest rate model
+    uint256 public constant blocksPerYear = 2102400;
+
+    /// @notice The stable base interest rate which is the y-intercept when utilization rate is 0(also known by base_premium)
+    uint256 public baseRatePerBlock;
+
+    /// @notice The premium rate applicable on optimal stable loan rate(also known by stable_rate_slope)
+    uint256 public stableRatePremium;
+
+    /// @notice The factor to be applied to the stable rate premium before adding to the interest rate
+    uint256 public optimalStableLoanRatio;
+
+    /// @notice The address of the owner, i.e. the Timelock contract, which can update parameters directly
+    address public owner;
+
+    /**
+     * @param baseRatePerYear_ The approximate target base APR, as a mantissa (scaled by BASE)
+     * @param stableRatePremium_ The multiplierPerBlock after hitting a specified utilization point
+     * @param optimalStableLoanRatio_ Optimal stable loan rate percentage.
+     * @param owner_ Address of the owner for this model(Governance)
+     */
+    constructor(
+        uint256 baseRatePerYear_,
+        uint256 stableRatePremium_,
+        uint256 optimalStableLoanRatio_,
+        address owner_
+    ) public {
+        owner = owner_;
+
+        updateStableRateModelInternal(baseRatePerYear_, stableRatePremium_, optimalStableLoanRatio_);
+    }
+
+    /**
+     * @notice Updates the parameters of the interest rate model (only callable by owner, i.e. Timelock)
+     * @param baseRatePerYear_ The approximate target base APR, as a mantissa (scaled by BASE)
+     * @param stableRatePremium_ The multiplierPerBlock after hitting a specified utilization point
+     * @param optimalStableLoanRatio_ Optimal stable loan rate percentage.
+     * @notice Emits NewStableInterestParams, after updating the parameters
+     * @notice Only governance
+     */
+    function updateStableRateModel(
+        uint256 baseRatePerYear_,
+        uint256 stableRatePremium_,
+        uint256 optimalStableLoanRatio_
+    ) external {
+        require(msg.sender == owner, "StableRateModel: only owner may call this function.");
+
+        updateStableRateModelInternal(baseRatePerYear_, stableRatePremium_, optimalStableLoanRatio_);
+    }
+
+    /**
+     * @notice Calculates the ratio of the stable borrows to total borrows
+     * @param stableBorrows The amount of stable borrows in the market
+     * @param totalBorrows The amount of total borrows in the market
+     * @return The stable loan rate as a mantissa between [0, BASE]
+     */
+    function stableLoanRatio(uint256 stableBorrows, uint256 totalBorrows) public pure returns (uint256) {
+        // Loan ratio is 0 when there are no stable borrows
+        if (totalBorrows == 0) {
+            return 0;
+        }
+
+        return (stableBorrows * BASE) / totalBorrows;
```

Comment:
Here we should use SafeMath. In solidity version 0.5.16, underflows/overflows don't revert.

Reply:
Fixed in [commit](https://github.com/VenusProtocol/venus-protocol/pull/210/commits/482b814841c2e68904e01250a89f48d5c745e83a)